sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableWriteIT.java

```diff
@@ -0,0 +1,196 @@
+
+    BigtableOptions.Builder bigtableOptionsBuilder = new BigtableOptions.Builder()
+        .setProjectId(options.getProjectId())
+        .setClusterId(options.getClusterId())
+        .setZoneId(options.getZoneId())
+        .setUserAgent("apache-beam-test");
+    bigtableOptions = bigtableOptionsBuilder.build();
+
+    session = new BigtableSession(bigtableOptions);
+    tableAdminClient = session.getTableAdminClient();
+  }
+
+  @Test
+  public void testE2EBigtableWrite() throws Exception {
+    final String tableName = bigtableOptions.getClusterName().toTableNameStr(tableId);
+    final String clusterName = bigtableOptions.getClusterName().toString();
+    final int numRows = 1000;
+    final List<KV<ByteString, ByteString>> testData = generateTableData(numRows);
+
+    createEmptyTable(clusterName, tableId);
+
+    Pipeline p = Pipeline.create(options);
+    p.apply(CountingInput.upTo(numRows))
+        .apply(ParDo.of(new DoFn<Long, KV<ByteString, Iterable<Mutation>>>() {
+          @Override
+          public void processElement(ProcessContext c) {
+            int index = c.element().intValue();
+
+            Iterable<Mutation> mutations =
+                ImmutableList.of(Mutation.newBuilder()
+                    .setSetCell(
+                        Mutation.SetCell.newBuilder()
+                            .setValue(testData.get(index).getValue())
+                            .setFamilyName(COLUMN_FAMILY_NAME))
+                    .build());
+            c.output(KV.of(testData.get(index).getKey(), mutations));
+          }
+        }))
+        .apply(BigtableIO.write()
+          .withBigtableOptions(bigtableOptions)
+          .withTableId(tableId));
+    p.run();
+
+    // Test number of column families and column family name equality
+    Table table = getTable(tableName);
+    assertThat(table.getColumnFamilies().keySet(), Matchers.hasSize(1));
+    assertThat(table.getColumnFamilies(), Matchers.hasKey(COLUMN_FAMILY_NAME));
+
+    // Test table data equality
+    List<KV<ByteString, ByteString>> tableData = getTableData(tableName);
+    assertThat(tableData, Matchers.containsInAnyOrder(testData.toArray()));
+  }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////
+  /** Helper function to generate KV test data. */
+  private List<KV<ByteString, ByteString>> generateTableData(int numRows) {
+    List<KV<ByteString, ByteString>> testData = new ArrayList<>(numRows);
+    for (int i = 0; i < numRows; ++i) {
+      ByteString key = ByteString.copyFromUtf8(String.format("key%09d", i));
+      ByteString value = ByteString.copyFromUtf8(String.format("value%09d", i));
+      testData.add(KV.of(key, value));
+    }
+
+    return testData;
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    final String tableName = bigtableOptions.getClusterName().toTableNameStr(tableId);
+    deleteTable(tableName);
```

Comment:
close the session, too. Anything other connections, etc., that we set up and should tear down?